### PR TITLE
tests: Expand mptcpd start/stop test.

### PR DESCRIPTION
**Pull request by ossama-othman**
Test mptcpd start and stop when MPTCP is both disabled and enabled.

### CODE REVIEW

File: tests/test-start-stop [hunk 1]
```diff
@@ -3,15 +3,70 @@
...
+old_value=`sysctl $variable | sed -e 's/.*= \(.*\)$/\1/'`
+
+disable_value=0
+enable_value=1
+
+# "Disabled" value is always 0, but the "enabled" value could be 1 or
+# 2, depending on the kernel in use.  For example, the
+# multipath-tcp.org kernel "enabled" value could be 1 or 2.  Use the
+# old "enabled" value in case it is 2 instead of 1.
+if [ $old_value -ne 0 ]
```
Comment (mjmartineau) [inline tests/test-start-stop:23]: My system complains here:

`tests/test-start-stop: line 23: [: too many arguments`

**Review by mjmartineau — COMMENTED** [review]
Seems to be a bash syntax issue

File: tests/test-start-stop (hunk 1 above)
Comment (ossama-othman) [reply tests/test-start-stop:26]: Fixed by commit ef905e7.

The regex didn't correctly handle multiple `net.mptcp` sysctl variables with `enabled` in them.  In particular, the 5.13 kernel on my Ubuntu box only has `net.mptcp.enabled` whereas the 5.15 kernel, for example, also has a `net.mptcp.checksum_enabled` variable, resulting multiple values being assigned to the `$old_value` shell variable.

**Comment by ossama-othman** [comment]
Expected content of `tests/test-start-stop.log`:

### Interactive `sudo` required
```
Running non-interactive sudo check...
sudo: a password is required
Running test without root privileges
PASS test-start-stop (exit status: 0)
```

### Non-interactive `sudo` allowed
```
Running non-interactive sudo check...
    succeeded
MPTCP is not enabled in the kernel.
Try 'sysctl -w net.mptcp.enabled=1'.
Required kernel MPTCP support not available.
TEST CASE: net.mptcp.enabled = 0 ...passed
TEST CASE: net.mptcp.enabled = 1 ...passed
PASS test-start-stop (exit status: 0)
```

**Review by mjmartineau — APPROVED** [review]
Regex fix works for me, I see the expected output in both sudo and non-sudo cases.

**Pull request merged**